### PR TITLE
Fix versions list order.

### PR DESCRIPTION
**Pull request by isoos**
- fixes the order on the package versions page
- fixes the order on the versions .json API endpoint

### CODE REVIEW

**Review by jonasfj — APPROVED** [review]
Curious, any reason not to change the ordering in the cached values instead?

I guess we would have to review all places it's used to see if it's okay. But did we check that other usages don't also expect a different ordering?

File: app/lib/frontend/handlers/package.dart [hunk 1]
```diff
@@ -76,7 +79,10 @@ Future<shelf.Response> packageVersionsListHandler(
       final dartSdkVersion = await getDartSdkVersion();
       return renderPkgVersionsPage(
         data,
-        versions.versions,
+        // The cached version list contains the versions in semantically ascending order.
+        // By reversing that list, we will have them in semantically descending order,
+        // which is the preferred order for displaying on the versions page.
```
Comment (jonasfj) [inline app/lib/frontend/handlers/package.dart:84]: This should also be documented in the documentation comment for `packageBackend.listVersionsCached`, especially if it promises to always return the versions in said order.

**Comment by isoos** [comment]
> Curious, any reason not to change the ordering in the cached values instead?

The cached value originally was for the `/api/packages/<pkg>` endpoint, and I didn't want to change that. Does the `pub` client have any assumption on the order of the versions?

> I guess we would have to review all places it's used to see if it's okay. But did we check that other usages don't also expect a different ordering?

Maybe we could use a different field names that returns it in specific orders, it would also help to address the comment about documentation. I'll update the PR.

File: app/lib/frontend/handlers/package.dart (hunk 1 above)
Comment (isoos) [reply app/lib/frontend/handlers/package.dart]: It is documented on the `versions` field when we are accessing, I think that's a better place for it, but we can also repeat it at other places.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**